tests/python_client/testcases/test_collection.py

```diff
@@ -2725,3 +2710,117 @@ def test_load_release_after_collection_drop(self, connect, collection):
         error = {ct.err_code: 0, ct.err_msg: "HasPartition failed: can\'t find collection: %s" % name}
         partition_w.load(check_task=CheckTasks.err_res, check_items=error)
         partition_w.release(check_task=CheckTasks.err_res, check_items=error)
+
+
+class TestCollectionString(TestcaseBase):
+    """
+    ******************************************************************
+      The following cases are used to test about string 
+    ******************************************************************
+    """
+
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_collection_string_field_is_primary(self):
+        """
+        target: test create collection with string field
+        method: 1. create collection with string field and vector field
+                2. set string fields is_primary=True
+        expected: Create  collection Successfully
+        """
+        self._connect()
+        c_name = cf.gen_unique_str(prefix)
+        int_field = cf.gen_int64_field()
+        vec_field = cf.gen_float_vec_field()
+        string_field = cf.gen_string_field(is_primary=True)
+        schema = cf.gen_collection_schema(fields=[int_field, string_field, vec_field])
```

Comment:
why does not use function "gen_default_collection_schema" to replace the above 4 lines for schema generation?

Reply:
Thanks for the suggestion has been modified